### PR TITLE
fix: fetch before attempting to checkout commit

### LINKED ISSUE #5645: Pull requests scans failing as of semgrep version 0.101.0 on git checkout
Hi  @spencerdrak  As per https://github.com/returntocorp/semgrep/issues/5562

I am also facing the same issue on most of the repos now with ```**git checkout** ```

using
```
name: Semgrep
on:
  pull_request: {}
  push:
    branches:
      - master
      - main
    paths:
      - .github/workflows/semgrep.yml
  schedule:
    - cron: '25 1 * * 4'
jobs:
  semgrep:
    name: Static Analysis Scan
    runs-on: 2xsmall
    env:
      SEMGREP_APP_TOKEN: ${{ secrets.SEMGREP_APP_TOKEN }}
    container:
      image: returntocorp/semgrep
    if: (github.actor != 'dependabot[bot]')
    steps:
      - uses: actions/checkout@v3
      - run: semgrep ci
```

Error
```
Run semgrep ci
Scan environment:
  versions    - semgrep 0.101.0 on python 3.10.5
  environment - running in environment github-actions, triggering event is pull_request
  semgrep.dev - authenticated as team
Fixing git state for github action pull request
Not on head ref: 0a2a0ad2ea131c4[6](https://github.com/team/infra/runs/7086714865?check_suite_focus=true#step:4:7)95[7](https://github.com/team/infra/runs//7086714865?check_suite_focus=true#step:4:8)c75d2[8](https://github.com/team/infra/runs//7086714865?check_suite_focus=true#step:4:9)35c6bc[9](https://github.com/team/infra/runs//7086714865?check_suite_focus=true#step:4:10)3b422869; checking that out now.
Command '['git', 'checkout', '0a2a0ad2ea131c46957c75d2835c6bc93b422869']' returned non-zero exit status 128.
Traceback (most recent call last):
  File "/usr/local/lib/python3.[10](https://github.com/team/infra/runs/7086714865?check_suite_focus=true#step:4:11)/site-packages/semgrep/commands/wrapper.py", line 35, in wrapper
    func(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/semgrep/commands/ci.py", line 306, in ci
    with fix_head_if_github_action(metadata):
  File "/usr/local/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.10/site-packages/semgrep/commands/ci.py", line 104, in fix_head_if_github_action
    checkout = subprocess.run(
  File "/usr/local/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['git', 'checkout', '0a2a0ad2ea131c46957c75d2835c6bc93b422869']' returned non-zero exit status [12](https://github.com/team/infra/runs/7086714865?check_suite_focus=true#step:4:13)8.
Error: Process completed with exit code 2.
```

Can this be fixed quicker as we have a lot of repos integrated to semgrep and it will start failing on 1000s of repos.

### PR DESCRIPTION
**Pull request by spencerdrak**
This address Issue #5645 , we will be testing and then releasing a bugfix version.

PR checklist:

- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 
 ## Unreleased
 
+### Fixed
+
+- `semgrep ci`: CI runs failed to checkout the commit assoociated with the head branch, and is fixed here.
```
Comment (brendongo) [inline CHANGELOG.md]:
```suggestion
- `semgrep ci`: CI runs in GithubActions failed to checkout the commit assoociated with the head branch, and is fixed here.
```

**Review by brendongo — APPROVED** [review]

**Comment by brendongo** [comment]
Approved pending testing built artifact

File: cli/src/semgrep/meta.py [hunk 2]
```diff
@@ -244,12 +244,32 @@ def _get_latest_commit_hash_in_branch(self, branch_name: str) -> str:
...
 
         Assumes we are in PR context
         """
         head_branch_name = self._head_branch_ref
         commit = self.glom_event(T["pull_request"]["head"]["sha"])
-        logger.debug(f"head branch ({head_branch_name}) has latest commit {commit}")
+        logger.debug(
+            f"head branch ({head_branch_name}) has latest commit {commit}, fetching that commit now."
+        )
+        process = subprocess.run(
```
Comment (underyx) [inline cli/src/semgrep/meta.py]: We have a helper function, `git_check_output`, which prints a nice error message if git is not available; we should use that here.

File: cli/src/semgrep/meta.py [hunk 3]
```diff
@@ -244,12 +244,32 @@ def _get_latest_commit_hash_in_branch(self, branch_name: str) -> str:
...
+            [
+                "git",
+                "fetch",
+                "origin",
+                "--force",
+                "--depth=1",
+                f"{commit}",
```
Comment (underyx) [inline cli/src/semgrep/meta.py]:
```suggestion
                commit,
```

Comment (spencerdrak) [reply cli/src/semgrep/meta.py]: changed

File: cli/src/semgrep/meta.py (hunk 2 above)
Comment (spencerdrak) [reply cli/src/semgrep/meta.py]: changed

**Pull request merged**